ctlra/ctlra.c

```diff
@@ -79,37 +85,64 @@ int ctlra_impl_dev_get_by_vid_pid(struct ctlra_t *ctlra, int32_t vid,
 }
 
 
-struct ctlra_dev_t *ctlra_dev_connect(struct ctlra_t *ctlra, int dev_id,
+struct ctlra_dev_t *ctlra_dev_connect(struct ctlra_t *ctlra,
+				      ctlra_dev_connect_func connect,
 				      ctlra_event_func event_func,
 				      void *userdata, void *future)
 {
-	if(dev_id < 0)
-		return 0;
-
-	if((unsigned)dev_id < CTLRA_NUM_DEVS && devices[dev_id].connect) {
-		struct ctlra_dev_t *new_dev = 0;
-		new_dev = devices[dev_id].connect(event_func,
-						  userdata,
-						  future);
-		if(new_dev) {
-			new_dev->ctlra_context = ctlra;
-			new_dev->dev_list_next = 0;
-
-			// if list empty, add as main ptr
-			if(ctlra->dev_list == 0) {
-				ctlra->dev_list = new_dev;
-				return new_dev;
-			}
-
-			// skip to end of list, and append
-			struct ctlra_dev_t *dev_iter = ctlra->dev_list;
-			while(dev_iter->dev_list_next)
-				dev_iter = dev_iter->dev_list_next;
-			dev_iter->dev_list_next = new_dev;
+	struct ctlra_dev_t *new_dev = 0;
```

Comment:
Probably a remnant of previous code changes. I'm not super pedantic - just pragmatically pedantic about code stuffs like this... you're right - this could be cleaner but just initializing correctly first time, or just not initializing and writing to later.